dnf-behave-tests/dnf/steps/repo.py

```diff
@@ -372,3 +378,21 @@ def given_system(context, system):
 def given_no_osrelease(context):
     behave.use_fixture(osrelease_fixture, context)
     context.scenario.osrelease.delete()
+
+
+@behave.step("I invalidate solvfile version of \"{path}\"")
+def rewrite_solvfile_version(context, path):
+    path = path.format(context=context)
+    ensure_file_exists(path)
+    fp = open(path, "r+b")
+    # Seek 40 bytes from start of file to the beginning userdata section
+    fp.seek(40, 0)
+    userdata_dnf_magic = fp.read(4)
+    assert userdata_dnf_magic == b'\x00dnf'
```

Comment:
These `assert`s are actually pretty confusing. Seeing them in test results I wondered what's going on and why isn't there a better message. Thing is one expects assert failures in test suites to be test result asserts, this one is asserting base test data before modifying them. Could you extend this with a message clearly saying this failed on mocking test data due to unexpected solv file content?

Reply:
Right, I have added messages to both asserts, thanks! :+1: